### PR TITLE
apps sc: expose external db values for harbor

### PR DESCRIPTION
**Pull request by robinAwallace**
**What this PR does / why we need it**:

Expose external db settings for harbor 

**Which issue this PR fixes**: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [x] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Review by crssnd — APPROVED** [review]

**Comment by robinAwallace** [comment]
I will do some updates after migration to yq4 is merged

**Comment by Xartos** [comment]
@robinAwallace Could you instead point this PR to the yq4 branch? It's much easier to review if I don't see the other things

File: helmfile/50-applications.yaml [hunk 1]
```diff
@@ -454,7 +454,7 @@ releases:
     component: backup
   chart: ./charts/harbor-backup
   version: 0.1.0
-  installed: {{ and .Values.harbor.enabled .Values.harbor.backup.enabled }}
+  installed: {{ and .Values.harbor.enabled (and .Values.harbor.backup.enabled (eq .Values.harbor.database.type "internal")) }}
```
Comment (Xartos) [inline helmfile/50-applications.yaml:457]: Isn't it a little confusing that if `harbor.backup.enabled` is enabled and the database type is set to external that it doesn't take a backup? In that case I would like to have an error telling be that this is not a valid configuration.

Comment (robinAwallace) [reply helmfile/50-applications.yaml:457]: Yes that is a good point. I will add a warning for it.

Comment (robinAwallace) [reply helmfile/50-applications.yaml]: Added the fail warning now

File: migration/v0.24.x-v0.25.x/upgrade-apps.md [hunk 2]
```diff
@@ -14,6 +14,11 @@
 
     ```console
     migration/v0.24.x-v0.25.x/migrate-harbor-redis-variables.sh
+
```
Comment (aarnq) [inline migration/v0.24.x-v0.25.x/upgrade-apps.md:17]:
```suggestion
    migration/v0.24.x-v0.25.x/migrate-harbor-redis-variables.sh
    ```
```

**Review by aarnq — APPROVED** [review]

**Pull request merged**